### ISSUE TITLE
Hard-coded URLs (e.g. GO.xrf_abbs) affected by root domain change

### ISSUE THREAD

**Issue by kltm** [owlcollab/owltools#276]
In some places (e.g. ./OWLTools-Annotation/src/main/java/owltools/gaf/rules/go/GoAnnotationRulesFactoryImpl.java), owltools seems to use some old-style URLs, like for http://www.geneontology.org/doc/GO.xrf_abbs . These should either use some kind of PURL or be written in a more flexible way.

**Comment by kltm** [comment]
Keeping @cmungall in the know, but see geneontology/operations ticket.

**Comment by kltm** [comment]
Also http://www.geneontology.org/quality_control/annotation_checks/annotation_qc.xml

**Comment by kltm** [comment]
While the document is now homed at: 
https://raw.githubusercontent.com/owlcollab/owltools/master/docs/legacy/annotation_qc.xml , we are still getting an error like:

```
2019-03-17 15:15:36,471 ERROR (AnnotationRulesFactoryImpl:84) Unable to load document form: http://www.geneontology.org/quality_control/annotation_checks/annotation_qc.xml
org.jdom.input.JDOMParseException: Error on line 1 of document http://geneontology.org/quality_control/annotation_checks/annotation_qc.xml: White spaces are required between publicId and systemId.
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:530)
	at org.jdom.input.SAXBuilder.build(SAXBuilder.java:905)
	at owltools.gaf.rules.AnnotationRulesFactoryImpl.init(AnnotationRulesFactoryImpl.java:82)
...
	at owltools.cli.CommandRunnerBase.run(CommandRunnerBase.java:68)
	at owltools.cli.CommandLineInterface.main(CommandLineInterface.java:12)
Caused by: org.xml.sax.SAXParseException; systemId: http://geneontology.org/quality_control/annotation_checks/annotation_qc.xml; lineNumber: 1; columnNumber: 50; White spaces are required between publicId and systemId.
	at org.apache.xerces.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)
	at org.apache.xerces.util.ErrorHandlerWrapper.fatalError(Unknown Source)
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
...
Caused by: org.xml.sax.SAXParseException; systemId: http://geneontology.org/quality_control/annotation_checks/annotation_qc.xml; lineNumber: 1; columnNumber: 50; White spaces are required between publicId and systemId.
	at org.apache.xerces.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)
	at org.apache.xerces.util.ErrorHandlerWrapper.fatalError(Unknown Source)

```

The new home would seem to be identical to the previous versions of the file, we are continuing to get some kind of error.

This error can be reproduced with:
```bash
~/local/src/git/owltools/OWLTools-Runner/bin/owltools --log-warning /tmp/go-basic.owl --gaf /tmp/foo.gaf --createReport --gaf-report-file -owltools-check.txt --gaf-report-summary-file -summary.txt --gaf-prediction-file -prediction.gaf --gaf-prediction-report-file -prediction-report.txt --gaf-validation-unsatisfiable-module -incoherent.owl --experimental-gaf-prediction-file -prediction-experimental.gaf --experimental-gaf-prediction-report-file -prediction-experimental-report.txt --gaf-run-checks || echo 'errors found'
```
Odd, I wonder if the headers are the same between the former and current versions. Google doesn't lead much to an explanation as to what may have changed. May have to start making random changes to see if something works.

Tagging @cmungall

**Comment by kltm** [comment]
http://52.27.86.54/quality_control/annotation_checks/annotation_qc.xml (old) diffs to same as http://www.geneontology.org/quality_control/annotation_checks/annotation_qc.xml (new).

**Comment by kltm** [comment]
Trying with S3 bucket https://s3.amazonaws.com/go-public/metadata/annotation_qc.xml :
```
HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 36886
Content-Type: text/xml
Date: Mon, 18 Mar 2019 17:57:21 GMT
ETag: "a4a7ba2fbb6ad7da1de2ded9ab8080b7"
Last-Modified: Mon, 18 Mar 2019 17:56:37 GMT
Server: AmazonS3
x-amz-id-2: wwUuu4mevFgGgBcAO0aAcvBTJhyQI7MVCzPyVKMF16Aw7hQChLgTa/vMC+tDRRu1IQgr+f8Khao=
x-amz-request-id: 86C234A41E741807
```

**Comment by kltm** [comment]
Still same error found.
@cmungall Is it possible that owltools cannot handle redirects?

**Comment by cmungall** [comment]
Likely the java lib used does not follow redirects by default

**Comment by kltm** [comment]
Choices now would seem to be:

1. change the owltools code/lib to accept redirects (or remove the need for the external file completely and just use one embedded locally)
2. to directly embed the file into geneontology.github.io at the right location as a static file

1 is the right solution, 2 may be easier to get out the door immediately.

**Comment by dougli1sqrd** [comment]
https://github.com/owlcollab/owltools/pull/279

**Comment by kltm** [comment]
This seems to be fixed with #279

**Comment by pgaudet** [comment]
Great!
